File: transport_marchandises_web/src/repository/VehiculeRepository.js
import ApiService from "../service/ApiService";
import ApiRoutes from "../routes/ApiRoutes.js";
import { Vehicule } from "../models/Vehicule";

class VehiculeRepository {
  constructor(apiService = ApiService) {
    this.apiService = apiService;
    this.pathVehicule = ApiRoutes.vehicule;
  }

  // 🔹 Ajouter un véhicule
  async postVehicule(vehicule) {
    console.log("Vehicule est ",vehicule)
    const dataResponse = await this.apiService.post(
      `${this.pathVehicule}`,
      vehicule.toJSON()
    );
    console.log("data Responce  est",dataResponse)
    return Vehicule.fromJSON(dataResponse);
  }

  // 🔹 Récupérer tous les véhicules
  async getVehiculesProfile() {
    const data = await this.apiService.get(`${this.pathVehicule}`);
    return data.map(Vehicule.fromJSON);
  }

  // 🔹 Récupérer un véhicule par ID
  async getVehicule(id) {
    const data = await this.apiService.get(`${this.pathVehicule}${id}/`);
    return Vehicule.fromJSON(data);
  }

  // 🔹 Modifier un véhicule
  async patchVehicule(id, data = {}) {
    const dataResponse = await this.apiService.patch(
      `${this.pathVehicule}${id}/`,
      data
    );
    return Vehicule.fromJSON(dataResponse);
  }

  // 🔹 Supprimer un véhicule
  async deleteVehicule(id) {
    const data = await this.apiService.delete(`${this.pathVehicule}${id}/`);
    return data;
  }
}

export default new VehiculeRepository();

File: transport_marchandises/lib/Repository/vehicule_repository.dart
import 'dart:ui';
import '../Service/Service.dart';
import '../models/utilisateur.dart' as User;
import '../models/vehicule.dart' as Vehicule;
import '../routes/routes.dart';

class VehiculeRepository {
  final ApiService apiService;
  VehiculeRepository(this.apiService);
  final String pathVehicule = Routes.vehicule;
  // add Vehicule
  Future<Vehicule.Vehicules?> postVehicule(Vehicule.Vehicules vehicule) async {
    final dataResponse = await apiService.post(
      "${pathVehicule}/",
      data: vehicule.toJSON(),
    );
    return Vehicule.Vehicules.jsonToVehicule(dataResponse);
  }

  Future<List<dynamic>?> getVehiculesProfile() async {
    final data = await apiService.get("${pathVehicule}/");
    // print(data.runtimeType);
    return data;
  }

  // get Vehicule par id
  Future<Vehicule.Vehicules?> getVehicule(int id) async {
    final data = await apiService.get("${pathVehicule}/${id}/");
    return Vehicule.Vehicules.jsonToVehicule(data);
  }

  Future<Vehicule.Vehicules?> patchVehicule(
    int id, {
    Map<String, dynamic>? data,
  }) async {
    print("id est ${id} et date est $data");
    final dataResponse = await apiService.patch(
      "${pathVehicule}/${id}/",
      data: data,
    );
    return Vehicule.Vehicules.jsonToVehicule(dataResponse);
  }

  Future<dynamic> delete(int id) async {
    final data = await apiService.delete("${pathVehicule}/${id}/");
    return data;
  }
}
